src/ints/mouse_dos_driver.cpp

```diff
@@ -0,0 +1,1440 @@
+// ***************************************************************************
+
+// Write and read directly to the screen. Do no use int_setcursorpos (LOTUS123)
+extern void WriteChar(uint16_t col, uint16_t row, uint8_t page, uint8_t chr,
+                      uint8_t attr, bool useattr);
+extern void ReadCharAttr(uint16_t col, uint16_t row, uint8_t page, uint16_t *result);
+
+void RestoreCursorBackgroundText()
+{
+    if (state.hidden || state.inhibit_draw)
+        return;
+
+    if (state.background) {
+        WriteChar(state.backposx,
+                  state.backposy,
+                  real_readb(BIOSMEM_SEG, BIOSMEM_CURRENT_PAGE),
+                  state.backData[0],
+                  state.backData[1],
+                  true);
+        state.background = false;
+    }
+}
+
+void DrawCursorText()
+{
+    // Restore Background
+    RestoreCursorBackgroundText();
+
+    // Check if cursor in update region
+    auto x = GetPosX();
+    auto y = GetPosY();
+    if ((y <= state.update_region_y[1]) && (y >= state.update_region_y[0]) &&
+        (x <= state.update_region_x[1]) && (x >= state.update_region_x[0])) {
+        return;
+    }
+
+    // Save Background
+    state.backposx = static_cast<uint16_t>(x >> 3);
+    state.backposy = static_cast<uint16_t>(y >> 3);
+    if (state.mode < 2)
+        state.backposx = static_cast<uint16_t>(state.backposx >> 1);
+
+    // use current page (CV program)
+    uint8_t page = real_readb(BIOSMEM_SEG, BIOSMEM_CURRENT_PAGE);
+
+    if (state.cursor_type == MouseCursor::Software) {
+        uint16_t result;
+        ReadCharAttr(state.backposx, state.backposy, page, &result);
+        state.backData[0] = (uint8_t)(result & 0xff);
+        state.backData[1] = (uint8_t)(result >> 8);
+        state.background  = true;
+        // Write Cursor
+        result = (result & state.text_and_mask) ^ state.text_xor_mask;
+        WriteChar(state.backposx,
+                  state.backposy,
+                  page,
+                  (uint8_t)(result & 0xff),
+                  (uint8_t)(result >> 8),
+                  true);
+    } else {
+        uint16_t address = static_cast<uint16_t>(page *
+            real_readw(BIOSMEM_SEG, BIOSMEM_PAGE_SIZE));
+        address = static_cast<uint16_t>(address +
+                (state.backposy * real_readw(BIOSMEM_SEG, BIOSMEM_NB_COLS) +
+                    state.backposx) * 2);
+        address /= 2;
+        uint16_t cr = real_readw(BIOSMEM_SEG, BIOSMEM_CRTC_ADDRESS);
+        IO_Write(cr, 0xe);
+        IO_Write(static_cast<io_port_t>(cr + 1),
+                 static_cast<uint8_t>((address >> 8) & 0xff));
+        IO_Write(cr, 0xf);
+        IO_Write(static_cast<io_port_t>(cr + 1),
+                 static_cast<uint8_t>(address & 0xff));
+    }
+}
+
+// ***************************************************************************
+// Graphic mode cursor
+// ***************************************************************************
+
+static uint8_t gfxReg3CE[9];
```

Comment:
gfx_reg_0x3ce perhaps?

Reply:
Changed to use names similar to defines from `int10.h`, encapsulated them in a struct.